src/js/apps/discovery/router.js

```
@@ -90,7 +90,9 @@ define([
 
       },
 
-      search: function (query) {
+      search: function (query, widgetName) {
```

calling NAVIGATE with "show-" + widgetName seems very reasonable to me; because when it is not there, the error gets raised -- but the logic under review here is different, it happens only for results-page (and silently ignores some other cases; that is not better)